File: 프로그래머스/0/120887. k의 개수/k의 개수.java
class Solution { // k의 개수
    public static int solution(int i, int j, int k) {
        int answer = 0;

        String target = String.valueOf(k); 
        
        for (int num = i; num <= j; num++) {
            String strNum = String.valueOf(num); 
            
            for (char ch : strNum.toCharArray()) { 
                if (String.valueOf(ch).equals(target)) {
                    answer++;
                }
            }
        }
        
        return answer;
    }

    public static void main(String[] args) {
        System.out.println(solution(1, 13, 1)); 
}
}

File: 프로그래머스/0/120887. k의 개수/k의 개수.go
func solution(i int, j int, k int) int {
	answer := 0

	for num := i; num <= j; num++ {
		n := num
		for n > 0 {
			if n%10 == k { 
				answer++
			}
			n /= 10 
		}
	}

	return answer
}